### PR TITLE
RND-809 Introduced Overlay component

### PR DESCRIPTION
**Pull request by Vorbert-Kruk**
## Description

Created `Overlay` React component based on [vanilla implementation of DDS Overlay](https://vanilla.delldesignsystem.com/2.29.0/index.html?path=/story/components-overlay--basic).

### Default
image

### Blurred
image

###  With position `absolute`
image

## Checklist

- [x] My code follows the [UI Code Style](https://cloudifysource.atlassian.net/l/c/x8fq902p).
- [x] I followed the [UI Testing Guidelines](https://cloudifysource.atlassian.net/l/cp/udCVfvrx) and verified that all tests/checks have passed.
- [x] I verified if that change requires any update in the [documentation](https://cloudifysource.atlassian.net/l/c/4XKtPocy).
- [x] I added proper labels to this PR.

## Tests

 

Jest snapshots and loki screenshots were updated.

## Documentation

Storybook stories were introduced.

### CODE REVIEW

File: src/components/dds/Overlay/Overlay.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+     */
+    variant?: OverlayVariant;
+    /**
+     * Css position value used by component
+     */
+    position?: OverlayPosition;
+    /**
+     * Controls if background scroll should be possible
+     */
+    disableBackgroundScroll?: boolean;
```
Comment (Vorbert-Kruk) [inline src/components/dds/Overlay/Overlay.tsx:25]: ### Note
Those props were inspired by [Overlay helpers description](https://vanilla.delldesignsystem.com/2.29.0/index.html?path=/docs/components-overlay--basic#overlay-helpers)

File: src/components/dds/Overlay/Overlay.tsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    /**
+     * Visibility of the component
+     */
+    visible?: boolean;
+    /**
+     * Visual effect applied to the overlay
+     */
+    variant?: OverlayVariant;
+    /**
+     * Css position value used by component
```
Comment (qooban) [inline src/components/dds/Overlay/Overlay.tsx:19]: Minor:

```suggestion
     * CSS position value used by component
```

File: src/components/dds/Overlay/Overlay.tsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+import classNames from 'classnames';
+
+export type OverlayVariant = 'default' | 'blur';
+export type OverlayPosition = 'fixed' | 'absolute';
+
+export interface OverlayProps extends React.HTMLAttributes<HTMLDivElement> {
+    /**
+     * Visibility of the component
+     */
+    visible?: boolean;
```
Comment (qooban) [inline src/components/dds/Overlay/Overlay.tsx:13]: As it determines visibility, shouldn't it be required? 🤔

File: src/components/dds/Overlay/Overlay.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,71 @@
+import React from 'react';
```
Comment (qooban) [inline src/components/dds/Overlay/Overlay.stories.tsx:1]: Just FYI I see that we have similar issue with Loki images like with `Modal` component.
Ref.: https://cloudifysource.atlassian.net/browse/RD-1146

No action needed, just sharing knowledge.

File: src/components/dds/Overlay/Overlay.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+export const basic: OverlayStory = () => {
+    const [isVisible, setIsVisible] = React.useState(false);
+
+    const handleButtonClick = () => {
+        setIsVisible(true);
+    };
+
+    const handleOverlayClick = () => {
+        setIsVisible(false);
+    };
```
Comment (qooban) [inline src/components/dds/Overlay/Overlay.stories.tsx:25]: Optional:

To save some lines, we can have custom hook exposing something like [isVisible, handleButtonClick, handleOverlayClick] and reuse it in 3 stories.

File: src/components/dds/Overlay/Overlay.tsx (hunk 3 above)
Comment (Vorbert-Kruk) [reply src/components/dds/Overlay/Overlay.tsx:13]: I wasn't sure, but yeah we can make it required

File: src/components/dds/Overlay/Overlay.stories.tsx (hunk 5 above)
Comment (Vorbert-Kruk) [reply src/components/dds/Overlay/Overlay.stories.tsx]: Sure, nice idea 💪 
Addressed within 9e60416bf7b5ad2e332d0e05ca99a606c46dd668

File: src/components/dds/Overlay/Overlay.tsx (hunk 3 above)
Comment (Vorbert-Kruk) [reply src/components/dds/Overlay/Overlay.tsx]: Addressed within 9e60416bf7b5ad2e332d0e05ca99a606c46dd668

Comment (qooban) [reply src/components/dds/Overlay/Overlay.tsx]: If you're wondering about the reasoning on my side - I was not sure if we should provide a component which won't be rendered with default props.

Alternatively you can keep it as optional, but make the default value to be `true`.

Comment (Vorbert-Kruk) [reply src/components/dds/Overlay/Overlay.tsx]: Oh,ok
I wasn't aware about the reasoning (thanks for sharing it), but idea looked quite promising

There are almost no cases when we'd want this component to be uncontrolled, so marking `visible` prop as required seems to make sense

**Review by qooban — APPROVED** [review]

**Pull request merged**